pkg/slo-controller/noderesource/plugins/resourceamplification/plugin.go

```diff
@@ -79,26 +91,35 @@ func (p *Plugin) Reset(node *corev1.Node, message string) []framework.ResourceIt
 	return nil
 }
 
+// Calculate calculates resource amplification ratio.
+// cpuAmplificationRatio = amplificationStrategy.CPURatio * CPUNormalizationRatio
+// memoryAmplificationRatio = amplificationStrategy.MemoryRatio
 func (p *Plugin) Calculate(_ *configuration.ColocationStrategy, node *corev1.Node, _ *corev1.PodList, _ *framework.ResourceMetrics) ([]framework.ResourceItem, error) {
-	normRatio, err := extension.GetCPUNormalizationRatio(node)
+	cpuAmplificationRatio, memoryAmplificationRatio := getAmplificationRatio(node)
```

Comment:
Could we modify the logic to make it more general (not limited to CPU and memory)? Only the CPU ratio shall be processed specifically and other resources can just be passed as is.